### PR TITLE
[WIP] Update grpc and gogoproto deps

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR updates the grpc and gogo/protobuf deps to match recent
updates to the envoyproxy/go-data-plane repo. This brings Istio
up-to-date with recent updates.

After updating the deps, I ran `go generate ./...` on the files.

I found a circular dep in the generation of pilot config.  That should be addressed somewhere.

I also note lots of struct initializations without field names. That requires manual attention.

Reviewers: I have not validated this update with any local testing.
For this reason, I have applied the "WIP" label.

/hold

### CODE REVIEW

**Comment by douglas-reid** [comment]
So, as far as I can tell, the generation mechanism for `pilot/pkg/config/kube/types.go` depends on that exact file for its generation purposes. Awesome-sauce.

**Comment by PiotrSikora** [comment]
Looks that you didn't actually update gRPC in `Gopkg.lock`.

Also, don't you need to update `go-control-plane` as well? Otherwise, there will be a mismatch between gRPC and protobuf from this PR and those used in the current version of `go-control-plane`.

**Comment by douglas-reid** [comment]
@PiotrSikora good points -- probably. i definitely haven't finished this PR.

**Comment by rshriram** [comment]
This needs to go into release-1.0.. We have to fix a gRPC streaming bug in Envoy (by setting the idle timeout to 0), but since @kyessenov updated go-control-plane's gRPC version, we need to update it here as well. Please rebase this PR to 1.0 branch.. This is pre-requisite for #7428 which is a release blocker (as gRPC streaming for long running connections is broken)

**Comment by rshriram** [comment]
update: I reverted the gRPC changes and have used a different version of go-control-plane without grpc/gogo changes (but with new data plane APIs). https://github.com/istio/istio/blob/release-1.0/Gopkg.toml#L112

All tests passed. So, when you fix up this PR, please make sure to update Gopkg.toml

**Comment by kyessenov** [comment]
@PiotrSikora I have no idea why dep fails to update gRPC even when it is constrained to a higher version already. Beats me.

**Comment by PiotrSikora** [comment]
@kyessenov don't you need to manually update the `Gopkg.lock` with the new commit and then run `dep ensure` to update `vendor` directory? That's what I've been doing in the past and it worked pretty well so far.

**Comment by kyessenov** [comment]
@PiotrSikora so we're back to manually picking SHAs? I thought lock file was not meant to be touched by humans.

File: Gopkg.toml [hunk 1]
```diff
@@ -285,7 +285,7 @@ ignored = [
 
 [[constraint]]
   name = "google.golang.org/grpc"
-  version = "1.11.1"
+  version = "^1.12.0"
```
Comment (PiotrSikora) [inline Gopkg.toml]: @kyessenov I think I found the issue, this should be `^1.13.0`.

**Comment by PiotrSikora** [comment]
@kyessenov maybe? But if you look at the commit history of `Gopkg.*`, that's definitely not the case.

**Comment by douglas-reid** [comment]
All, should I continue with this PR?  I feel like it probably still valuable, but having been OOO for awhile, I've lost track of current status / desires.

**Comment by kyessenov** [comment]
ref #7985

**Comment by rshriram** [comment]
please continue with the PR.. we need this in order to be able to pull in other updates from Envoy.

**Comment by douglas-reid** [comment]
Closing in favor of https://github.com/istio/istio/pull/8036

**Pull request closed without merging**